src/main/java/frc/robot/subsystems/PoseEstimatorSubsystem.java

```diff
@@ -23,36 +23,49 @@
+        double ta = Robot.LIMELIGHT_SUBSYSTEM_ONE.getTa();
+        double ta2 = Robot.LIMELIGHT_SUBSYSTEM_TWO.getTa();
+        Pose2d firstLimelightPose = Robot.LIMELIGHT_SUBSYSTEM_ONE.getGyroBasedRobotPose();
+        Pose2d secondLimelightPose =  Robot.LIMELIGHT_SUBSYSTEM_TWO.getGyroBasedRobotPose();
         Pose2d robotPose = Robot.POSE_ESTIMATOR_SUBSYSTEM.getCurrentPose();
-        Robot.LIMELIGHT_SUBSYSTEM.getPureLimelightRobotPose();
+        Robot.LIMELIGHT_SUBSYSTEM_ONE.getPureLimelightRobotPose();
         Robot.DRIVE_TRAIN_SUBSYSTEM.getPose();
         Robot.DRIVE_TRAIN_SUBSYSTEM.getGyroscopeRotation();
         Robot.DRIVE_TRAIN_SUBSYSTEM.getSwerveModulePositions();
 
-        boolean hasTarget = Robot.LIMELIGHT_SUBSYSTEM.getTv() == 1;
-        boolean isWithinXDistance = Math.abs(robotPose.getX() - limelightPose.getX()) < 2;
-        boolean isWithinYDistance = Math.abs(robotPose.getY() - limelightPose.getY()) < 2;
+        boolean hasTarget = Robot.LIMELIGHT_SUBSYSTEM_ONE.getTv() == 1;
+        boolean hasTarget2 = Robot.LIMELIGHT_SUBSYSTEM_TWO.getTv() == 1;
+        boolean firstLimelightIsWithinXDistance = Math.abs(robotPose.getX() - firstLimelightPose.getX()) < 2;
+        boolean firstLimelightIsWithinYDistance = Math.abs(robotPose.getY() - firstLimelightPose.getY()) < 2;
+        boolean secondLimelightIsWithinXDistance = Math.abs(robotPose.getX() - secondLimelightPose.getX()) < 2;
+        boolean secondLimelightIsWithinYDistance = Math.abs(robotPose.getY() - secondLimelightPose.getY()) < 2;
         boolean targetAreaIsSufficient = ta >= 0.2;
-        boolean fiducialIdIsCorrect = LimelightHelpers.getFiducialID("limelight") <= 8;
-
+        boolean targetAreaIsSufficient2 = ta2 >= 0.2;
+        boolean fiducialIdIsCorrect2 = LimelightHelpers.getFiducialID("firstLimelight") <= 8;
+        boolean fiducialIdIsCorrect = LimelightHelpers.getFiducialID("secondlimelight") <= 8;
+        
         // Scale the confidence of the vision estimate by how much ApilTag we see.
         double inverseArea = 1 - ta;
         double inverseCubed = Math.pow(inverseArea, 3);
         double clampedCube = Math.max(inverseCubed, Constants.MINIMUM_VISION_STANDARD_DEVIATION);
 
         visionStdDevs = VecBuilder.fill(clampedCube, clampedCube, clampedCube);
 
-        if (hasTarget && isWithinXDistance && isWithinYDistance && targetAreaIsSufficient && fiducialIdIsCorrect) {
-            m_poseEstimator.addVisionMeasurement(limelightPose, timeStamp, visionStdDevs);
+        if (hasTarget && firstLimelightIsWithinXDistance && firstLimelightIsWithinYDistance && targetAreaIsSufficient && fiducialIdIsCorrect) {
+            m_poseEstimator.addVisionMeasurement(firstLimelightPose, _timeStamp, visionStdDevs);
+        }
+
+
+        if (hasTarget2 && secondLimelightIsWithinXDistance && secondLimelightIsWithinYDistance && targetAreaIsSufficient2 && fiducialIdIsCorrect2) {
```

Comment:
I think a method on LimelightSubsystem called `HasVisionTarget` would clean up a lot of things that are pulled into this subsystem. If it's valid then you can call `getGyroBasedRobotPose` to get the pose for `addVisionMeasurement`.